### ISSUE TITLE
Conda 4.2.10, 4.2.11 error when parsing .condarc with S3 channel URL

### ISSUE THREAD

**Issue by oinksoft** [conda/conda#3783]
We've recently run into issues using our S3 conda channel. We actually found this because of `boto` upgrading `conda` from installed, stable versions to latest (4.2.11). This forces us to pin all of our conda environments to conda 4.2.9 (described at bottom). I looked at #3774 but adding a key prefix to the end of the bucket URL has no effect (this would be unsuitable in any case, but wanted to see if the issue was related).

```
(root) vagrant-ubuntu-trusty-64>> conda --version
conda 4.2.11
```

```
(root) vagrant-ubuntu-trusty-64>> conda search test
An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Traceback (most recent call last):
  File "/home/vagrant/mc/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/cli/main.py", line 152, in main
    return conda_exception_handler(_main)
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/exceptions.py", line 497, in conda_exception_handler
    print_unexpected_error_message(e)
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/exceptions.py", line 443, in print_unexpected_error_message
    info_stdout, info_stderr = get_info()
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/exceptions.py", line 413, in get_info
    args.func(args, p)
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/cli/main_info.py", line 225, in execute
    channels = list(prioritize_channels(channels).keys())
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/models/channel.py", line 380, in prioritize_channels
    channel = Channel(chn)
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/models/channel.py", line 161, in __call__
    c = Channel.from_value(value)
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/models/channel.py", line 211, in from_value
    return Channel.from_url(value)
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/models/channel.py", line 196, in from_url
    return parse_conda_channel_url(url)
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/models/channel.py", line 132, in parse_conda_channel_url
    configured_token) = _read_channel_configuration(scheme, host, port, path)
  File "/home/vagrant/mc/lib/python2.7/site-packages/conda/models/channel.py", line 122, in _read_channel_configuration
    return (Url(host=host, port=port).url.rstrip('/'), path.strip('/') or None,
AttributeError: 'NoneType' object has no attribute 'strip'
```

~/.condarc:

```yml
channels:
  - defaults
  - s3://our-conda-channel-bucket
  - conda-forge
```

To work around this issue:

* Comment out lines with S3 channel URLs in .condarc
* `conda install conda=4.2.9`
* Add `conda ==4.2.9` to $CONDA_ENV/conda-meta/pinned (or else other packages will upgrade it)
* Uncomment lines with S3 channel URLs in .condarc

**Comment by kalefranz** [comment]
@oinksoft We released a conda patch version `4.2.12` yesterday.  Can you update and see if the problem persists?  If so, can you give me the new stack trace?  It's possible the line numbers have changed.

**Comment by oinksoft** [comment]
@kalefranz Thanks, this works as expected in 4.2.12!
